### PR TITLE
Disambiguate features in `gstd`

### PR DESCRIPTION
**Pull request by breathx**
- Removed `std` feature
- Added support for `String` in program's metadata and custom payloads in `gtest`
- Sending payload function now requires generics, which implements `AsRef<[u8]>`. This allows us to don't convert something into bytes by ourselves. Previous variant is still available.
- Fixed bug with `bail!` macro, which appeared after splitting into `gcore` and `gstd` libs. He was throwing compile error due to wrong function execution.
- `gstd::prelude` a bit expanded 

@gear-tech/dev

### CODE REVIEW

**Review by NikVolf — CHANGES_REQUESTED** [review]
This is totally against design of gstd (which should be an analogue of rust std)

File: gstd/src/prelude.rs [hunk 1]
```diff
@@ -16,14 +16,32 @@
...
+    string::{String, ToString},
+    vec,
+    vec::Vec,
+};
+
+#[cfg(feature = "std")]
+pub use std::{
+    borrow::ToOwned,
+    boxed::Box,
+    collections::{BTreeMap, VecDeque},
```
Comment (shamilsan) [inline gstd/src/prelude.rs]: `std` just reimports these entities from `alloc`. Should we double the code?

**Review by NikVolf — APPROVED** [review]

File: gstd/src/prelude.rs (hunk 1 above)
Comment (breathx) [reply gstd/src/prelude.rs]: `std` feature already removed

**Comment by Anfid** [comment]
Great changes, but it would force std to all users who use gstd only with alloc. I would prefer if std could still be behind a feature, even if we enable it by default.

**Review by ukint-vs — APPROVED** [review]

**Pull request merged**